File: backend/src/main/java/zw/co/hisolutions/core/service/PreRequisiteService.java
package zw.co.hisolutions.core.service;
  
import zw.co.hisolutions.common.service.GenericService;
import zw.co.hisolutions.core.entity.PreRequisite;

public interface PreRequisiteService extends GenericService<PreRequisite, Long>{
    public PreRequisite findByName(String name);   
}

File: frontend/src/app/admin/services/rest/pre-requisite.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http'; 

import {RestDataService} from 'shared/services/rest-data.service';

@Injectable()
export class  PreRequisiteService extends RestDataService {

    constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/prerequisites'; 
        super(httpClient, resourcePath);
    }

}
